### PR TITLE
[Release 3.0] Fix Async thread blocking on SqlConnection open for AAD modes

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Ports #1213 to v3.0
from @Kaur-Parminder

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx [hunk 1]
```diff
@@ -4614,4 +4614,4 @@
   <data name="ArgumentOutOfRange_NeedNonNegNum" xml:space="preserve">
     <value>Non-negative number required.</value>
   </data>
-</root>
+</root>
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx:4617]: Revert this change, please.

File: src/Microsoft.Data.SqlClient/netcore/src/Resources/Strings.Designer.cs [hunk 2]
```diff
@@ -2833,7 +2824,7 @@ internal class Strings {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Error code 0x{0}; state {1}.
+        ///   Looks up a localized string similar to Error code 0x{0}.
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/netcore/src/Resources/Strings.Designer.cs:2827]: Other changes except this one are unrelated in this file.

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs [hunk 3]
```diff
@@ -9685,7 +9676,7 @@ internal class Strings {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Error code 0x{0}; state {1}.
+        ///   Looks up a localized string similar to Error code 0x{0}.
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs:9679]: Other changes except this one are unrelated in this file.

**Comment by DavoudEshtehari** [comment]
Since this includes a breaking change, could it be backported as a hotfix?

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx (hunk 1 above)
Comment (Kaur-Parminder) [reply src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx]: Davoud, I have addressed this change.

File: src/Microsoft.Data.SqlClient/netcore/src/Resources/Strings.Designer.cs (hunk 2 above)
Comment (Kaur-Parminder) [reply src/Microsoft.Data.SqlClient/netcore/src/Resources/Strings.Designer.cs:2836]: Davoud, I have addressed this change.

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs (hunk 3 above)
Comment (Kaur-Parminder) [reply src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs:9688]: Davoud, I have addressed this change.

**Comment by cheenamalhotra** [comment]
> Since this includes a breaking change, could it be backported as a hotfix?

I agree, we should skip error message updates, and only fix the async calls here.
